File: codegen/codegen-erlang/src/test/resources/ir/sigv4_sign.expected.erl
-spec sign(client_config(), Operation :: atom(), http_request()) -> http_request().
sign(Config, Operation, Request) ->
    Credentials = maps:get(credentials, Config),
    Region = maps:get(region, Config, <<"us-east-1">>),
    Service = maps:get(signing_name, Config),
    Unsigned = maps:get({unsigned_payload, Operation}, Config, false),
    Opts = #{
        unsigned_payload => Unsigned,
        endpoint_host => endpoint_host_from_config(Config)
    },
    sign_request(Request, Credentials, Region, Service, Opts).

File: codegen/codegen-elixir/src/test/resources/ir/sigv4_sign.expected.ex
@spec sign(map(), atom(), RuntimeTypes.HttpRequest.t()) :: RuntimeTypes.HttpRequest.t()
def sign(config, operation, request) do
  credentials = Map.fetch!(config, :credentials)
  region = Map.get(config, :region, "us-east-1")
  service = Map.fetch!(config, :signing_name)
  unsigned = Map.get(config, {:unsigned_payload, operation}, false)
  opts =
    %{
      unsigned_payload: unsigned,
      endpoint_host: Utils.endpoint_host_from_config(config)
    }
  sign_request(request, credentials, region, service, opts)
end
